pandas/io/common.py

```diff
@@ -391,13 +398,17 @@ def _get_handle(path_or_buf, mode, encoding=None, compression=None,
         elif encoding:
             # Python 3 and encoding
             f = open(path_or_buf, mode, encoding=encoding)
-        else:
+        elif is_text:
             # Python 3 and no explicit encoding
             f = open(path_or_buf, mode, errors='replace')
+        else:
+            # Python 3 and binary mode
+            f = open(path_or_buf, mode)
         handles.append(f)
 
     # in Python 3, convert BytesIO or fileobjects passed with an encoding
-    if compat.PY3 and (compression or isinstance(f, need_text_wrapping)):
+    if compat.PY3 and is_text and\
```

Comment:
Is there an opportunity to simplify things? What is the relationship between `need_text_wrapping`, `compression`, and `is_text` --- I will think about this, but I think currently, the logic for when `TextIOWrapper`  gets applied is confusing.